snow/networking/sender/sender.go

```diff
@@ -665,9 +665,10 @@ func (s *sender) SendGetAncestors(ctx context.Context, nodeID ids.NodeID, reques
 		s.engineType,
 	)
 
-	// Sending a GetAncestors to myself always fails.
+	// Sending a GetAncestors to myself will fail. To avoid constantly sending
+	// myself requests when not connected to any peers, we rely on the timeout
+	// firing to deliver the GetAncestorsFailed message.
 	if nodeID == s.ctx.NodeID {
```

Comment:
Could this introduce a performance regression when syncing the X-chain DAG?

Reply:
The X-chain DAG is synced with the current validator set. So this is only potentially an issue if the node that is syncing is a validator... which shouldn't really be happening